linkerd/app/outbound/src/logical.rs

```diff
@@ -64,10 +76,7 @@ impl svc::Param<SkipHttpDetection> for Logical<()> {
 
 impl<P> Logical<P> {
     pub fn addr(&self) -> Addr {
-        self.logical_addr
-            .as_ref()
-            .map(|LogicalAddr(a)| Addr::from(a.clone()))
-            .unwrap_or_else(|| self.orig_dst.0.into())
+        Addr::from(self.logical_addr.clone().0)
```

Comment:
this is only used in one place now, probably best to just inline rather than have a dedicated helper

Reply:
Hmm, I don't think that's true, this is used in a bunch of places?
![Screenshot_20210414_093823](https://user-images.githubusercontent.com/2796466/114746966-2967d500-9d05-11eb-9b27-15f4fc8b16ed.png)

We definitely could remove it, though, specially since it's much simpler now, but it would be a larger diff.